presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/TestHiveFileSystemAbfs.java

```diff
@@ -0,0 +1,198 @@
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import io.prestosql.plugin.hive.AbstractTestHive.Transaction;
+import io.prestosql.plugin.hive.azure.HiveAzureConfig;
+import io.prestosql.plugin.hive.azure.PrestoAzureConfigurationInitializer;
+import io.prestosql.spi.connector.ColumnMetadata;
+import io.prestosql.spi.connector.ConnectorTableMetadata;
+import io.prestosql.spi.connector.SchemaTableName;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Parameters;
+import org.testng.annotations.Test;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.prestosql.plugin.hive.HiveTableProperties.BUCKETED_BY_PROPERTY;
+import static io.prestosql.plugin.hive.HiveTableProperties.BUCKET_COUNT_PROPERTY;
+import static io.prestosql.plugin.hive.HiveTableProperties.EXTERNAL_LOCATION_PROPERTY;
+import static io.prestosql.plugin.hive.HiveTableProperties.SKIP_FOOTER_LINE_COUNT;
+import static io.prestosql.plugin.hive.HiveTableProperties.SKIP_HEADER_LINE_COUNT;
+import static io.prestosql.plugin.hive.HiveTableProperties.SORTED_BY_PROPERTY;
+import static io.prestosql.plugin.hive.HiveTableProperties.STORAGE_FORMAT_PROPERTY;
+import static io.prestosql.spi.type.BigintType.BIGINT;
+import static java.lang.String.format;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+import static org.testng.util.Strings.isNullOrEmpty;
+
+public class TestHiveFileSystemAbfs
+        extends AbstractTestHiveFileSystem
+{
+    private String container;
+    private String account;
+    private String accessKey;
+
+    @Parameters({
+            "hive.hadoop2.metastoreHost",
+            "hive.hadoop2.metastorePort",
+            "hive.hadoop2.databaseName",
+            "hive.hadoop2.azure-wasb-container",
+            "hive.hadoop2.azure-wasb-account",
+            "hive.hadoop2.azure-wasb-access-key"})
+    @BeforeClass
+    public void setup(String host, int port, String databaseName, String container, String account, String accessKey)
+    {
+        checkArgument(!isNullOrEmpty(host), "expected non empty host");
+        checkArgument(!isNullOrEmpty(databaseName), "Expected non empty databaseName");
+        checkArgument(!isNullOrEmpty(container), "expected non empty container");
+        checkArgument(!isNullOrEmpty(account), "expected non empty account");
+        checkArgument(!isNullOrEmpty(accessKey), "expected non empty accessKey");
+
+        this.container = container;
+        this.account = account;
+        this.accessKey = accessKey;
+
+        super.setup(host, port, databaseName, false, createHdfsConfiguration());
+    }
+
+    @Override
+    protected void onSetupComplete()
+    {
+        ensureTableExists(table, "presto_test_external_fs_v2", ImmutableMap.of());
```

Comment:
nit: i was able to remove `_v2` suffix in https://github.com/prestosql/presto/commit/e364b6b493fb60734a11332e727eae91b829b707

(no change requested now)